### PR TITLE
[Enhancement] Add storage-related profiles for load

### LINKED ISSUE #46373: [Enhancement] Add storage-related profiles for load
Need storage metrics in the load profile to know why the load is slow if the bottleneck is at the storage

### PR DESCRIPTION
**Pull request by banmoy**
## Why I'm doing:
Need storage metrics in the load profile to know why the load is slow if the bottleneck is at the storage

## What I'm doing:
* For primary/peer tablets, there are following metrics
   * the number/cost of methods for DeltaWriter write/close/commit
   * the statistics about memtable including the number/cost  of sort, agg, flush
   * for primary tablets, also includes the time to sync segments to secondary replica
   * the pending time of tasks in the queue such as async delta writer task, flush task, segment replica task
* For secondary tablets, there are following metrics
   * the number/cost of methods for DeltaWriter add_segment/commit
   * the statistics about segment flush

Fix #46373

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [X] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/runtime/load_channel.cpp [hunk 1]
```diff
@@ -331,6 +331,16 @@ Status LoadChannel::_deserialize_chunk(const ChunkPB& pchunk, Chunk& chunk, fast
     return Status::OK();
 }
 
+std::vector<int64_t> LoadChannel::_get_index_ids() {
+    std::vector<int64_t> index_ids;
+    std::lock_guard l(_lock);
```
Comment (imay) [inline be/src/runtime/load_channel.cpp:336]: How often is this fiction called? Will this impact the performance of concurrent ingestions

**Comment by banmoy** [comment]
@mergify rerun

File: be/src/storage/segment_replicate_executor.h [hunk 2]
```diff
@@ -74,6 +74,14 @@ class ReplicateChannel {
     Status _st = Status::OK();
 };
 
+struct SegmentReplicateStat {
+    int32_t num_pending_tasks = 0;
+    int32_t num_executing_tasks = 0;
```
Comment (wyb) [inline be/src/storage/segment_replicate_executor.h:79]:
```suggestion
    int32_t num_running_tasks = 0;
```

File: be/src/storage/segment_replicate_executor.cpp [hunk 3]
```diff
@@ -229,7 +246,11 @@ Status ReplicateToken::submit(std::unique_ptr<SegmentPB> segment, bool eos) {
         return Status::InternalError(fmt::format("{} segment=null eos=false", debug_string()));
     }
     auto task = std::make_shared<SegmentReplicateTask>(this, std::move(segment), eos);
-    return _replicate_token->submit(std::move(task));
+    Status st = _replicate_token->submit(std::move(task));
+    if (st.ok()) {
+        _stat.num_pending_tasks += 1;
```
Comment (wyb) [inline be/src/storage/segment_replicate_executor.cpp:251]: should use atomic variable

Comment (banmoy) [reply be/src/storage/segment_replicate_executor.cpp]: Fixed

File: be/src/storage/segment_replicate_executor.h (hunk 2 above)
Comment (banmoy) [reply be/src/storage/segment_replicate_executor.h]: Fixed

File: be/src/storage/memtable_flush_executor.cpp [hunk 4]
```diff
@@ -83,11 +87,13 @@ class MemtableFlushTask final : public Runnable {
...
     std::function<void(std::unique_ptr<SegmentPB>, bool)> _cb;
+    int64_t _create_time_ns;
 };
 
 std::ostream& operator<<(std::ostream& os, const FlushStatistic& stat) {
-    os << "(flush time(ms)=" << stat.flush_time_ns / 1000 / 1000 << ", flush count=" << stat.flush_count << ")"
-       << ", flush flush_size_bytes = " << stat.flush_size_bytes;
+    os << "(flush time(ms)=" << stat.memtable_stats.flush_time_ns / 1000 / 1000 << ", flush count=" << stat.flush_count
+       << ")"
+       << ", flush flush_size_bytes = " << stat.memtable_stats.flush_memory_size;
```
[the remark this answers is not in the document]
Comment (banmoy) [reply be/src/storage/memtable_flush_executor.cpp]: Fixed

**Review by wyb — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

File: be/src/storage/segment_replicate_executor.cpp [hunk 5]
```diff
@@ -215,7 +232,11 @@ Status ReplicateToken::submit(std::unique_ptr<SegmentPB> segment, bool eos) {
         return Status::InternalError(fmt::format("{} segment=null eos=false", debug_string()));
     }
     auto task = std::make_shared<SegmentReplicateTask>(this, std::move(segment), eos);
-    return _replicate_token->submit(std::move(task));
+    Status st = _replicate_token->submit(std::move(task));
+    if (st.ok()) {
+        _stat.num_pending_tasks.fetch_add(1, std::memory_order_relaxed);
```
Comment (trueeyu) [inline be/src/storage/segment_replicate_executor.cpp:237]: Why add `num_pending_tasks` when submit task failed?

**Review by trueeyu — APPROVED** [review]

**Pull request merged**